src/ostree.jsx

```diff
@@ -484,10 +520,213 @@ const DeploymentDetails = (akey, info, packages, doRollback, doUpgrade, doRebase
     });
 };
 
+const DeploymentActions = ({ deploymentIndex, deploymentIsPinned, isCurrent, isStaged }) => {
+    const [isKebabOpen, setKebabOpen] = useState(false);
+
+    const togglePin = () => {
+        const pinFlags = [];
+        if (deploymentIsPinned) {
+            pinFlags.push("--unpin");
+        }
+
+        cockpit.spawn(["ostree", "admin", "pin", ...pinFlags, deploymentIndex], { superuser: "try" })
+                .then(() => setKebabOpen(false));
+    };
+
+    const deleteDeployment = () => {
+        cockpit.spawn(["ostree", "admin", "undeploy", deploymentIndex], { superuser: "try" })
+                .then(() => setKebabOpen(false));
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-403-20231128-144720-91f0c34a-fedora-coreos-devel//Coverage/lcov/github-pr.diff.gcov.html)